### DISCUSSION TITLE
Use of conductivity value for gold

### DISCUSSION THREAD

**Discussion by kalosu** [NanoComp/meep#2326] [Q&A]
Hello there again MEEP community, 

I came across with the issue: https://github.com/NanoComp/meep/issues/1084 and I wonder if for the case of gold, the use of memory could be improved if I define the material parameters of gold by setting the conductivity as specified in:   https://meep.readthedocs.io/en/latest/Materials/#conductivity-and-complex

I have tried doing this but I get a warning with the indication that there exist an epsilon with a value smaller than 1 and that I should decrease the value of the Courant factor.

Since I was already using the standard meep library for Gold, I already had a Courant factor of 0.1

Despite this, my simulation just breaks down with the common message: "simulation fields are NaN or Inf"

I know that for negative permittivity values I should use a broadband Drude-Lorentzian susceptibility as indicated in:  https://meep.readthedocs.io/en/latest/FAQ/#how-do-i-import-n-and-k-values-into-meep but I was trying to go around this in order to improve the memory use as mentioned at the top, allowing me then to speed up my simulations.

Any idea about a possible work around?

**Comment by oskooi** [comment]
The material parameters for Au in the Meep materials library are based on fitting its $\varepsilon(\omega)$ to a Drude-Lorentzian susceptibility profile over the wavelength range of [0.2 - 12.4 μm](https://github.com/NanoComp/meep/blob/391e7c58862c3189401b2c85350315da9fed489d/python/materials.py#L291-L292). Because of this wide bandwidth, the fit involves 5 Lorentzian susceptibility terms. Each susceptibility term in turn requires three polarization fields to be stored over the entire cell and time stepped in the chunks containing the Au structure.

One possible approach to improving computational efficiency when modeling Au is to simply perform your own fit over a narrow bandwidth using this [script](https://github.com/NanoComp/meep/blob/master/python/examples/eps_fit_lorentzian.py). The fewer susceptibility terms you need to obtain an accurate fit, the less memory and floating operations necessary for storage and time stepping. 

(Note that a wavelength-independent $\varepsilon$ with negative real part will cause the fields to blow up as described in https://meep.readthedocs.io/en/latest/FAQ/#why-does-my-simulation-diverge-if-the-permittivity-is-less-than-0.)

**Reply by kalosu** [reply]
Hi @oskooi , 

I have tried your suggestion by limiting the number of susceptibility terms (2 in this case for a narrow bandwidth from around 0.700 to 1.2 microns. I am actually just interested in the response at an even narrower bandwidth close to 900 nm). 
Nevertheless I cannot get the simulation to work. At some point the field value just explodes and the simulation stops because of this.

Here you can see the parameters that are found by the optimization routine for the two susceptibility terms:

lorentzian susceptibility: frequency=1.54862, gamma=0
drude susceptibility: frequency=1, gamma=0.0626212

You can see that for the lorentzian case, the gamma parameter is 0. I imagine that this has an influence on the fact that the field does not decay and just builds up.

As for the drude susceptibility term, the frequency is always one which if I use the material function for gold already included within MEEP, the value of this parameter is really small: 1e-10

Any idea about what could be happening here and about what could I do in order to get the simulation to not blow out?

**Reply by oskooi** [reply]
Have you tried the suggestions in https://meep.readthedocs.io/en/latest/FAQ/#why-are-the-fields-blowing-up-in-my-simulation?

1. Replace `PML` with an `Absorber`.
2. Disable subpixel smoothing (set `eps_averaging=False` in the `Simulation` constructor).
3. Increase the `resolution`.
